src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result = {};
+  const values: T[keyof T][] = [];
+
+  items.forEach((item) => {
+    if (values.indexOf(item[key]) === -1) {
+      values.push(item[key]);
+    }
+  });
+
+  values.forEach((value) => {
+    result[value] = items.filter((item) => item[key] === value);
+  });
```

Comment:
```suggestion
  items.forEach((item) => {
    if (!result[item[key]])) {
      result[item[key]] = [];
    }
    
    result[item[key]].push(item);
  });
```

Можна сильно спростити рішення